src/fedify/mq/gcloud-pubsub-mq.ts

```diff
@@ -0,0 +1,105 @@
+import type { Message, PubSub } from '@google-cloud/pubsub';
+
+import type {
+    MessageQueue,
+    MessageQueueEnqueueOptions,
+    MessageQueueListenOptions,
+} from '@fedify/fedify';
+import type { Logger } from '@logtape/logtape';
+
+export class GCloudPubSubMessageQueue implements MessageQueue {
+    private pubSubClient: PubSub;
+    private topicIdentifier: string;
+    private subscriptionIdentifier: string;
+    private logger: Logger;
+
+    constructor(
+        pubSubClient: PubSub,
+        topicIdentifier: string,
+        subscriptionIdentifier: string,
+        logger: Logger,
+    ) {
+        this.pubSubClient = pubSubClient;
+        this.topicIdentifier = topicIdentifier;
+        this.subscriptionIdentifier = subscriptionIdentifier;
+        this.logger = logger;
+    }
+
+    async enqueue(
+        message: any,
+        options?: MessageQueueEnqueueOptions,
+    ): Promise<void> {
+        const delay = options?.delay?.total('millisecond') ?? 0;
+
+        this.logger.info(
+            `Enqueuing message [FedifyID: ${message.id}] with delay: ${delay}ms`,
+        );
+
+        if (delay > 0) {
```

Comment:
Do we know when/why we need to support a delay?

Reply:
Going off the docs → https://fedify.dev/manual/mq#implement-messagequeue-enqueue-method

I think this is an internal Fedify thing for the retries (exponential backoff)